fcrepo-client-impl/src/main/java/org/fcrepo/client/impl/FedoraRepositoryImpl.java

```diff
@@ -315,8 +327,117 @@ public boolean isWritable() {
+
+        } catch (final Exception e) {
+            LOGGER.error("could not encode URI parameter", e);
+            throw new FedoraException(e);
+        } finally {
+            post.releaseConnection();
+        }
+
+    }
+
+    @Override
+    public void commitTransaction() throws FedoraException {
+        final HttpPost post = httpHelper.createPostMethod(TX_ID.get()
+                + FCR_TX + FCR_COMMIT, null);
+        try {
+            final HttpResponse response = httpHelper.execute(post);
+            final String uri = post.getURI().toString();
+            final StatusLine status = response.getStatusLine();
+            final int statusCode = status.getStatusCode();
+
+            if (statusCode == SC_FORBIDDEN) {
+                LOGGER.error("request to commit transaction {} is not authorized.", uri);
+                throw new ForbiddenException("request to commit transaction " + uri + " is not authorized.");
+            } else {
+                if (statusCode != SC_NO_CONTENT) {
+                    LOGGER.error("error committing back transaction {}: {} {}", uri,
+                            statusCode, status.getReasonPhrase());
+                    throw new FedoraException("error committing transaction " + uri + ": " + statusCode + " " +
+                            status.getReasonPhrase());
+                }
+            }
+            LOGGER.debug("Committed transaction {}", TX_ID.get());
+        } catch (final Exception e) {
+            LOGGER.error("could not encode URI parameter", e);
+            throw new FedoraException(e);
+        } finally {
+            post.releaseConnection();
+            TX_ID.remove();
+        }
+    }
+
+    @Override
+    public void rollbackTransaction() throws FedoraException {
+        final HttpPost post = httpHelper.createPostMethod(TX_ID.get()
+                + FCR_TX + FCR_ROLLBACK, null);
+        try {
+            final HttpResponse response = httpHelper.execute(post);
+            final String uri = post.getURI().toString();
+            final StatusLine status = response.getStatusLine();
+            final int statusCode = status.getStatusCode();
+
+            if (statusCode == SC_FORBIDDEN) {
+                LOGGER.error("request to rollback transaction {} is not authorized.", uri);
+                throw new ForbiddenException("request to rollback transaction " + uri + " is not authorized.");
+            } else {
+                if (statusCode != SC_NO_CONTENT) {
+                    LOGGER.error("error rolling back transaction {}: {} {}", uri, statusCode, status.getReasonPhrase());
+                    throw new FedoraException("error rolling back transaction " + uri + ": " + statusCode + " " +
+                            status.getReasonPhrase());
+                }
+            }
+
+            LOGGER.debug("Rolled back transaction {}", TX_ID.get());
+        } catch (final Exception e) {
+            LOGGER.error("could not encode URI parameter", e);
```

Comment:
Please do not dump exception stacktraces into the log. Instead, print the `e.getMessage()`.